libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -23,9 +23,10 @@ proc init*(
 ): EntryRecord {.gcsafe, raises: [].} =
   EntryRecord(value: value, time: time.get(TimeStamp(ts: $times.now().utc)))
 
-type ReceivedTable = ref Table[PeerId, EntryRecord]
-
-type LocalTable* = Table[Key, EntryRecord]
+type
+  ReceivedTable = TableRef[PeerId, Opt[EntryRecord]]
```

Comment:
not sure about this: table should have value as type; not opt value. 
when there is value in table it's not optional. table { gabe=human }, key "gabe" has value.
Opt.None() is same as having no key=value in table, that is if you look for "vlado" in table, there is no value. adding additional level of value presence seems unnecessary ?